burstphoto/texture/texture.swift

```diff
@@ -653,38 +649,61 @@ func texture_like(_ in_texture: MTLTexture) -> MTLTexture {
+    command_encoder.setComputePipelineState(sum_rect_columns_float_state)
+    let thread_groups_per_grid = MTLSize(width: summed_y.width, height: summed_y.height, depth: 1)
+    let threads_per_thread_group = get_threads_per_thread_group(sum_rect_columns_float_state, thread_groups_per_grid)
+    
     command_encoder.setTexture(in_texture, index: 0)
-    command_encoder.setTexture(avg_y, index: 1)
-    command_encoder.dispatchThreads(threads_per_grid, threadsPerThreadgroup: threads_per_thread_group)
+    command_encoder.setTexture(summed_y, index: 1)
+    command_encoder.setBytes([0], length: MemoryLayout<Int32>.stride, index: 1)
+    command_encoder.setBytes([0], length: MemoryLayout<Int32>.stride, index: 2)
+    command_encoder.setBytes([in_texture.height], length: MemoryLayout<Int32>.stride, index: 3)
+    command_encoder.setBytes([Int32(mosaic_pattern_width)], length: MemoryLayout<Int32>.stride, index: 4)
+    command_encoder.dispatchThreads(thread_groups_per_grid, threadsPerThreadgroup: threads_per_thread_group)
+
+    // Sum along the row
+    let sum_buffer = device.makeBuffer(length: (mosaic_pattern_width*mosaic_pattern_width)*MemoryLayout<Float32>.size,
+                                       options: .storageModeShared)!
+    let threads_per_grid_x = MTLSize(width: mosaic_pattern_width, height: mosaic_pattern_width, depth: 1)
+    let threads_per_thread_group_x = get_threads_per_thread_group(sum_row_state, threads_per_grid_x)
+    command_encoder.setComputePipelineState(sum_row_state)
+    command_encoder.setTexture(summed_y, index: 0)
+    command_encoder.setBuffer(sum_buffer, offset: 0, index: 0)
+    command_encoder.setBytes([Int32(summed_y.width)],       length: MemoryLayout<Int32>.stride, index: 1)
+    command_encoder.setBytes([Int32(mosaic_pattern_width)], length: MemoryLayout<Int32>.stride, index: 2)
+    // TODO: Should this have the same number of threads? I think it should only be 1 thread.
+    command_encoder.dispatchThreads(threads_per_grid_x, threadsPerThreadgroup: threads_per_thread_group_x)
+    
+    // Calculate the average from the sum
+    let state       = per_sub_pixel ? divide_buffer_state                       : sum_divide_buffer_state
+    let buffer_size = per_sub_pixel ? mosaic_pattern_width*mosaic_pattern_width : 1
+    let avg_buffer  = device.makeBuffer(length: buffer_size*MemoryLayout<Float32>.size, options: .storageModeShared)!
+    // If doing per-subpixel, the total number of pixels of each subpixel is 1/(mosaic_pattern_withh)^2 times the total
+    let num_pixels_per_value = Float(in_texture.width * in_texture.height) / (per_sub_pixel ? Float(mosaic_pattern_width*mosaic_pattern_width) : 1.0)
+    let threads_per_grid_divisor = MTLSize(width: buffer_size, height: 1, depth: 1)
+    let threads_per_thread_group_divisor = get_threads_per_thread_group(state, threads_per_grid_divisor)
+    command_encoder.setComputePipelineState(state)
```

Comment:
Just for consistency, I would suggest to move this line up that it is before the let threads_per_gid_divisor = ... line